Desktop_Interface/isobufferbuffer.h

```diff
@@ -1,27 +1,59 @@
 #ifndef ISOBUFFERBUFFER_H
 #define ISOBUFFERBUFFER_H
 
-//isobufferbuffer is a buffer designed for getting the last n things added in reverse order, in O(1) time.
-
 #include <QDebug>
-#include <stdlib.h>
+#include <cstdint>
 #include <string>
+#include <memory>
 
-class isoBufferBuffer
-{
+/** @file isobufferbuffer.h
+ *  @brief This  module  implements  a  data structure that allows
+ *  insertion  of  single  characters  and  a  view  of the last N
+ *  inserted characters in constant time.
+ *
+ *  To  obtain  such  complexity,  a  double  ring buffer is used.
+ *  That  is,  two  identical  ring buffers are mantained adjacent
+ *  in  memory.  If we always return a pointer to the begging of a
+ *  range  that ends in the second buffer, we will always return a
+ *  valid  address(*),  even  when the requested length is greater
+ *  than  the  current position being inserted into in the buffer.
+ *
+ *  (*) By  valid  address  I  mean  that  both the addresses that
+ *  represent  the  begining and end of the requested query result
+ *  are within the allocated buffer.
+ */
+class isoBufferBuffer {
 public:
-    isoBufferBuffer(uint32_t length);
-    void add(uint8_t newByte);
-    void add(char newChar);
-    void add(std::string newString);
-    char *get(uint32_t length);
-    uint32_t getNumCharsInBuffer();
+	isoBufferBuffer ( uint32_t length );
```

Comment:
Don't know why you've added the double tabs here.

Reply:
Ah, that's weird. It doesn't look like that for me even though i use 4 space tabs.